frontend-react/src/components/Admin/EditSenderSettings.tsx

```diff
@@ -253,39 +254,31 @@ const EditSenderSettingsForm: React.FC<EditSenderSettingsFormProps> = ({
 };
 
 export type EditSenderSettingsProps = {
-    orgName: string;
-    senderName: string;
+    orgname: string;
+    sendername: string;
     action: "edit" | "clone";
 };
 
 export function EditSenderSettings() {
-    const { orgName, senderName, action } =
-        useParams<EditSenderSettingsProps>();
-    /* useParams now returns possible undefined. This will error up to a boundary
-     * if the url param is undefined */
-    if (
-        orgName === undefined ||
-        senderName === undefined ||
-        action === undefined
-    )
-        throw Error("Expected orgName & settingType from path");
+    const { orgname, sendername, action } =
+        useValidParams<EditSenderSettingsProps>([
```

Comment:
This PR fully implements a pattern seen in PR 1 with undefined params checks. `useValidParams` will take a string array of keys to check incoming params values are not undefined.

Reply:
The type is just piped right to `useParams` but that doesn't handle validating whether or not the param came through. The goal is to _not_ render when url params are missing, so just calling use params doesn't suffice, you need to throw if one is missing that's required for render. Because paths are **runtime** variables, we can't expect TS to handle their type assertions. 

Those prop types were leftover, actually, from before. I had removed them, but felt that may step on some toes, so I added them back in according to their definitions in `master`.

Reply:
I think an ideal footprint and use would be

```typescript
type UrlParams = { one: string, two: string }
const params = useValidParams<UrlParams>(["one"])
params.one // Required by hook so no need for optional chain
params?.two // Still possibly undefined, requires optional chain
```

and to rename the parameter of `useValidParams` to `requiredParams` to indicate that the reason for the array since in these it just matches the destructured response...that's definitely a weird way to call it.

Reply:
The above approach offers the ability to put our filter params in URLs now, with MUCH easier assertion, and that'll enable users to share links to filtered submissions or daily-data tables. That's where this idea was headed when I decided to hook-ify it.

Reply:
I'm still missing something. The call doesn't understand types and only checks for undefined. So if it's required, just make it required in the typedef. If it's optional, then the behavior when it's missing is custom based on the component. (e.g. Components with date ranges, the optional end date missing just means "use the current date".)

What happens when a required component parameter comes from a cgi parameter and it's malformed? Should there just be a default param used instead?

This might require a quick conversation to explain the real world use case? It would be nice to include @doug-s-nava and others so we can all understand the correct use case.

Reply:
Oh, I didn't clarify. UseParams returns everything as possibly undefined, independent of whatever type you pass in. 

Equal to Partial<T>. 

That's why the array is needed for checking required keys.

Reply:
Here's a quick screen grab of what that would look like.
<img width="614" alt="Screen Shot 2022-09-09 at 1 49 06 PM" src="https://user-images.githubusercontent.com/4022304/189413431-f59eb42c-0ad4-4f9a-b33b-3d56eddfc9b6.png">

Reply:
Updated the comment and parameter name to better describe its existence